wardrobe/kitties/src/lib.rs

```diff
@@ -500,28 +561,111 @@ impl TryFrom<&DynamicallyTypedData> for KittyData {
 
 impl SimpleConstraintChecker for FreeKittyConstraintChecker {
     type Error = ConstraintCheckerError;
-    /// Checks:
-    ///     - `input_data` is of length 2
-    ///     - `output_data` is of length 3
-    ///
+
     fn check(
         &self,
         input_data: &[DynamicallyTypedData],
         _peeks: &[DynamicallyTypedData],
         output_data: &[DynamicallyTypedData],
     ) -> Result<TransactionPriority, Self::Error> {
-        // Input must be a Mom and a Dad
-        ensure!(input_data.len() == 2, Self::Error::TwoParentsDoNotExist);
-
-        let mom = KittyData::try_from(&input_data[0])?;
-        let dad = KittyData::try_from(&input_data[1])?;
-        KittyHelpers::can_breed(&mom, &dad)?;
-
-        // Output must be Mom, Dad, Child
-        ensure!(output_data.len() == 3, Self::Error::NotEnoughFamilyMembers);
-
-        KittyHelpers::check_new_family(&mom, &dad, output_data)?;
+        match &self {
+            Self::Create => {
+                // Ensure that no inputs are being consumed.
+                ensure!(
+                    input_data.is_empty(),
+                    ConstraintCheckerError::CreatingWithInputs
+                );
+
+                // Ensure that at least one kitty is being created.
+                ensure!(
+                    !output_data.is_empty(),
+                    ConstraintCheckerError::CreatingNothing
+                );
+
+                // Ensure the outputs are the right type.
+                for utxo in output_data {
+                    let _utxo_kitty = utxo
+                        .extract::<KittyData>()
+                        .map_err(|_| ConstraintCheckerError::BadlyTyped)?;
+                }
+                Ok(0)
```

Comment:
Should we somehow check that a Kitty with the same DNA does not exist?
cc @JoshOrndorff

Reply:
This is one of the reasons I didn't like minting kitties from scratch

Reply:
I raised this question earlier with Joshy and found that no need to check for duplicate DNA check since there can be twin kitties with duplicate DNA. So I removed the duplicate DNA check.

Reply:
Well, we state in multiple parts of the documentation that DNA is unique 😄 

As far as I remember, the twins use-case was not initially part of Kitties, what's the reason behind adding it?

Moreover, twins in my opinion should not have the exact same DNA, as it is also in real life

Reply:
@muraca  
Please see the discussion here: https://github.com/Off-Narrative-Labs/Tuxedo/pull/171#discussion_r1509088145
Earlier I implemented the DNA check for create operation and other operations also where it can be abused.
There were 2 options: For simplicity, I chose the 2nd option i.e allow duplicate DNA kitties.

Reply:
On a call recently, I encouraged @NadigerAmit to design the game fully before trying to build it and get a PR approved. Specifically I encouraged him to consider:

1. Who is going to be able to mint in production? Only a centralized team? Or any rando? How will minting be sybil resistant?
2. Should every kitty have unique DNA? If so implement a universal creator pattern as shown in #185 .

I don't think there are right vs wrong answers. But you need a design and you need to be consistent about it. I worry we reached a point where Amit feels very "close" to getting this PR merged, but I feel the design work isn't even done to compare the code against.